### PR TITLE
ConfirmModal: Reuse confirm content

### LINKED ISSUE #88576: ConfirmModal: Reuse content
We want to be able to render confirmation content inside the drawer as it is in this design: https://www.figma.com/design/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?node-id=3966-9429&m=dev
Therefore, the Confirm Modal content is extracted into a new component so that it can be used anywhere

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

It extracts the ConfirmModal content into a new component so that it can be used anywhere

**Why do we need this feature?**

We need to show confirmation content in a Drawer and we want to be consistent with the ConfirmModal .
[Design](https://www.figma.com/design/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?node-id=3966-9429&m=dev)

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/88576

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmContent.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+export interface ConfirmContentProps {
+  /** Modal content */
+  body: React.ReactNode;
+  /** Modal description */
+  description?: React.ReactNode;
+  /** Text for confirm button */
+  confirmText: string;
+  /** Text for dismiss button */
+  dismissText?: string;
```
[the remark this answers is not in the document]
Comment (juanicabanas) [reply packages/grafana-ui/src/components/ConfirmModal/ConfirmContent.tsx:24]: got it, I can change it

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx [hunk 2]
```diff
@@ -1,53 +1,27 @@
...
-import { Input } from '../Input/Input';
-import { Box } from '../Layout/Box/Box';
-import { Stack } from '../Layout/Stack/Stack';
+import { ButtonVariant } from '../Button';
 import { Modal } from '../Modal/Modal';
 
-export interface ConfirmModalProps {
+import { ConfirmContent, ConfirmContentProps } from './ConfirmContent';
+
+export interface ConfirmModalProps extends Omit<ConfirmContentProps, 'fillToConfirmText'> {
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx:12]: I wouldn't extend the props interface here - i think it's fine to just redefine them :)

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmContent.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  /** Variant for dismiss button */
+  dismissVariant?: ButtonVariant;
+  /** Text user needs to fill in before confirming */
+  fillToConfirmText?: string;
+  /** Text for alternative button */
+  alternativeText?: string;
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/ConfirmModal/ConfirmContent.tsx:30]: Following Josh's comment, we could rename all the button props to be clearer (and maybe move them together)
`alternativeText` -> `alternativeButtonLabel`
`dismissVariant` -> `dismissButtonVariant`

**Comment by juanicabanas** [comment]
@joshhunt @JoaoSilvaGrafana suggestions applied :) Please, take a look again.
I don't think this is a breaking change PR, right?

**Review by joshhunt — APPROVED** [review]

**Review by joshhunt — APPROVED** [review]

**Pull request merged**